hooks/tk-multi-publish2/basic/publish_session.py

```diff
@@ -0,0 +1,733 @@
+                it will be unchecked. Optional, True by default.
+
+        :param settings: Dictionary of Settings. The keys are strings, matching
+            the keys returned in the settings property. The values are `Setting`
+            instances.
+        :param item: Item to process
+
+        :returns: dictionary with boolean keys accepted, required and enabled
+        """
+
+        path = _session_path()
+
+        if not path:
+            # the session has not been saved before (no path determined).
+            # provide a save button. the session will need to be saved before
+            # validation will succeed.
+            self.logger.warn(
+                "The Maya session has not been saved.",
+                extra=_get_save_as_action()
+            )
+
+        self.logger.info(
+            "Maya '%s' plugin accepted the current Maya session." %
+            (self.name,)
+        )
+        return {
+            "accepted": True,
+            "checked": True
+        }
+
+    def validate(self, settings, item):
+        """
+        Validates the given item to check that it is ok to publish. Returns a
+        boolean to indicate validity.
+
+        :param settings: Dictionary of Settings. The keys are strings, matching
+            the keys returned in the settings property. The values are `Setting`
+            instances.
+        :param item: Item to process
+        :returns: True if item is valid, False otherwise.
+        """
+
+        publisher = self.parent
+        path = _session_path()
+
+        if not path:
+            # the session still requires saving. provide a save button.
+            # validation fails.
+            self.logger.error(
+                "The Maya session has not been saved.",
+                extra=_get_save_as_action()
+            )
+            return False
+
+        # ensure we have an updated project root
+        project_root = cmds.workspace(q=True, rootDirectory=True)
+        item.properties["project_root"] = project_root
+
+        # warn if no project root could be determined.
+        if not project_root:
+            self.logger.warning(
+                "Your session is not part of a maya project.",
+                extra={
+                    "action_button": {
+                        "label": "Set Project",
+                        "tooltip": "Set the maya project",
+                        "callback": lambda: mel.eval('setProject ""')
+                    }
+                }
+            )
+
+        # get the path in a normalized state. no trailing separator,
+        # separators are appropriate for current os, no double separators,
+        # etc.
+        path = sgtk.util.ShotgunPath.normalize(path)
+
+        # if the session item has a known work file template, see if the path
+        # matches. if not, warn the user and provide a way to save the file to
+        # a different path
+        work_file_template = item.properties.get("work_file_template")
```

Comment:
New logic to warn if there is a template configured but the path doesn't match. It offers a "Save File" button in the log window that will launch wf2 to save (if configured).

Reply:
so awesome!